cores/arduino/wiring.c

```diff
@@ -22,44 +18,126 @@
 
 #include "wiring_private.h"
 
-// the prescaler is set so that timer0 ticks every 64 clock cycles, and the
+// CHANGE by Gijs Withagen, 18-02-2019
+// Prescaler can be changed in runtime and still have a correct millis()
+// the micros() just has a glitch, but counted form changing the prescaler
+// it gives proper relative values.
+//
+// Defalult the prescaler is set so that timer0 ticks every 64 clock cycles, and the
 // the overflow handler is called every 256 ticks.
-#define MICROSECONDS_PER_TIMER0_OVERFLOW (clockCyclesToMicroseconds(64 * 256))
```

Comment:
The 256 magic number can be changed by the user code, too.  Probably the same change is needed for the overflow handler: a `getTicksPerOverflow()` function.